icechunk/src/format/snapshot.rs

```diff
@@ -1,74 +1,51 @@
-use std::{collections::BTreeMap, convert::Infallible, sync::Arc};
+use std::{collections::BTreeMap, convert::Infallible, num::NonZeroU64, sync::Arc};
 
+use bytes::Bytes;
 use chrono::{DateTime, Utc};
 use err_into::ErrorInto;
 use flatbuffers::{FlatBufferBuilder, VerifierOptions};
 use itertools::Itertools as _;
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
 
-use crate::metadata::{
-    ArrayShape, ChunkKeyEncoding, ChunkShape, Codec, DataType, DimensionNames, FillValue,
-    StorageTransformer, UserAttributes,
-};
-
 use super::{
     flatbuffers::gen,
     manifest::{Manifest, ManifestExtents, ManifestRef},
     AttributesId, ChunkIndices, IcechunkFormatError, IcechunkFormatErrorKind,
-    IcechunkResult, ManifestId, NodeId, Path, SnapshotId, TableOffset,
+    IcechunkResult, ManifestId, NodeId, Path, SnapshotId,
 };
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct UserAttributesRef {
-    pub object_id: AttributesId,
-    pub location: TableOffset,
-}
-
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub enum UserAttributesSnapshot {
-    Inline(UserAttributes),
-    Ref(UserAttributesRef),
-}
-
-#[derive(Clone, Debug, Hash, PartialEq, Eq)]
-pub enum NodeType {
-    Group,
-    Array,
+pub struct DimensionShape {
+    array_length: u64,
```

Comment:
```suggestion
    dim_length: u64,
```

I think this would be less confusing in the long run. `array.size` means "number of elements in the array" in numpy, so I was confused at first.